internal/client/work_queues.go

```diff
@@ -0,0 +1,194 @@
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusCreated {
+		errorBody, _ := io.ReadAll(resp.Body)
+
+		return nil, fmt.Errorf("status code %s, error=%s", resp.Status, errorBody)
+	}
+
+	var queue api.WorkQueue
+	if err := json.NewDecoder(resp.Body).Decode(&queue); err != nil {
+		return nil, fmt.Errorf("failed to decode response: %w", err)
+	}
+
+	return &queue, nil
+}
+
+// List returns a list of work queue matching filter criteria.
+func (c *WorkQueuesClient) List(ctx context.Context, filter api.WorkQueueFilter) ([]*api.WorkQueue, error) {
+	var buf bytes.Buffer
+	if err := json.NewEncoder(&buf).Encode(&filter); err != nil {
+		return nil, fmt.Errorf("failed to encode filter: %w", err)
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, c.routePrefix+"/filter", &buf)
+	if err != nil {
+		return nil, fmt.Errorf("error creating request: %w", err)
+	}
+
+	setDefaultHeaders(req, c.apiKey)
+
+	resp, err := c.hc.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("http error: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		errorBody, _ := io.ReadAll(resp.Body)
+
+		return nil, fmt.Errorf("status code %s, error=%s", resp.Status, errorBody)
+	}
+
+	var queues []*api.WorkQueue
+	if err := json.NewDecoder(resp.Body).Decode(&queues); err != nil {
+		return nil, fmt.Errorf("failed to decode response: %w", err)
+	}
+
+	return queues, nil
+}
+
+// Get returns details for a work queue by name.
+func (c *WorkQueuesClient) Get(ctx context.Context, name string) (*api.WorkQueue, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, c.routePrefix+"/"+name, http.NoBody)
+	if err != nil {
+		return nil, fmt.Errorf("error creating request: %w", err)
+	}
+
+	setDefaultHeaders(req, c.apiKey)
+
+	resp, err := c.hc.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("http error: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		errorBody, _ := io.ReadAll(resp.Body)
+
+		return nil, fmt.Errorf("status code %s, error=%s", resp.Status, errorBody)
+	}
+
+	var queue api.WorkQueue
+	if err := json.NewDecoder(resp.Body).Decode(&queue); err != nil {
+		return nil, fmt.Errorf("failed to decode response: %w", err)
+	}
+
+	return &queue, nil
+}
+
+// Update modifies an existing work queue by name.
+func (c *WorkQueuesClient) Update(ctx context.Context, name string, data api.WorkQueueUpdate) error {
+	var buf bytes.Buffer
+	if err := json.NewEncoder(&buf).Encode(&data); err != nil {
+		return fmt.Errorf("failed to encode data: %w", err)
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPatch, c.routePrefix+"/"+name, &buf)
+	if err != nil {
+		return fmt.Errorf("error creating request: %w", err)
+	}
+
+	setDefaultHeaders(req, c.apiKey)
+
+	resp, err := c.hc.Do(req)
+	if err != nil {
+		return fmt.Errorf("http error: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusNoContent {
```

Comment:
just making sure - is 200 and 204 both a valid response code for PATCH /queue?

Reply:
updated!